### PR TITLE
Merge nanit-dev to nanit-master

### PR DESCRIPTION
**Pull request by tomer-algo**

### CODE REVIEW

File: demo/demo_inference.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+            csv_headers.extend([keypoint + '_x', keypoint + '_y', keypoint + '_conf'])
+
+        csv_output_filename = 'pose-data.csv'
+        with open(os.path.join(out_dir, csv_output_filename), 'w', newline='') as csvfile:
+            csvwriter = csv.writer(csvfile)
+            csvwriter.writerow(csv_headers)
+            #             csvwriter.writerow(['Detected {} out of {} images'.format(relevant_count, len(image_list))])
+            csvwriter.writerows(csv_output_rows)
+
+print('detected {} / {}'.format(relevant_count, count))
```
Comment (felixNan) [inline demo/demo_inference.py:247]: Add a simple function for inference on one file which will the file_pat, the output path and the baby and head bounding boxes.

File: lib/nanit_utils/gen_coco_data.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import os
```
Comment (felixNan) [inline lib/nanit_utils/gen_coco_data.py:1]: Consider the relevance of this file

File: lib/nanit_utils/gen_mpii_data.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
+import os
```
Comment (felixNan) [inline lib/nanit_utils/gen_mpii_data.py:1]: Consider the relevance of this file

File: lib/nanit_utils/extract_data_from_batches.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
+import os
```
Comment (felixNan) [inline lib/nanit_utils/extract_data_from_batches.py:1]: Consider the relevance of this file

File: lib/dataset/nanit_mpii.py [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+import imagesize
+import cv2
+from lib.nanit_utils.utils_functions import transform_box_xywh
+
+logger = logging.getLogger(__name__)
+
+
+class NanitMPIIDataset(JointsDataset):
+    def __init__(self, cfg, root, image_set, is_train, transform=None):
```
Comment (felixNan) [inline lib/dataset/nanit_mpii.py:28]: Why not use dataset instead of image_set?

File: lib/dataset/nanit_mpii.py [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+        super().__init__(cfg, root, image_set, is_train, transform, gen_mpii_joints_mapping_dict())
+
+        self.num_joints = 16
+        self.flip_pairs = [[0, 5], [1, 4], [2, 3], [10, 15], [11, 14], [12, 13]]
```
Comment (felixNan) [inline lib/dataset/nanit_mpii.py:32]: Use some joint mapping dict to make this part more readable without magic numbers.

File: lib/dataset/nanit_mpii.py [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+            image_name = a['image']
+
+            c = np.array(a['center'], dtype=np.float)
+            s = np.array([a['scale'], a['scale']], dtype=np.float)
+
+            # Adjust center/scale slightly to avoid cropping limbs
+            if c[0] != -1:
+                c[1] = c[1] + 15 * s[1]
+                s = s * 1.25
+
```
Comment (felixNan) [inline lib/dataset/nanit_mpii.py:94]: what is s and c, use more intuitive names

File: lib/dataset/nanit_mpii.py [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+        PCKh = np.ma.array(PCKh, mask=False)
+        PCKh.mask[6:8] = True
+
+        jnt_count = np.ma.array(jnt_count, mask=False)
+        jnt_count.mask[6:8] = True
+        jnt_ratio = jnt_count / np.sum(jnt_count).astype(np.float64)
+
+        name_value = [
+            ('Head', PCKh[head]),
```
Comment (felixNan) [inline lib/dataset/nanit_mpii.py:197]: Move to some eval utils module

File: lib/nanit_utils/utils_functions.py [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+import cv2
+
+def get_bbox_from_skeleton_points(points):
+    xmin, ymin = np.min(points, axis=0)
+    xmax, ymax = np.max(points, axis=0)
+    box_w = (xmax - xmin) * 1.05
+    box_h = (ymax - ymin) * 1.05
+    bbox = [xmin - 0.05 * box_w, ymin - 0.05 * box_h, box_w, box_h]
+    return bbox
+
```
Comment (felixNan) [inline lib/nanit_utils/utils_functions.py:11]: unused?

File: lib/nanit_utils/utils_functions.py [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+def get_bbox_xywh_center(bbox_xywh):
+    center = [bbox_xywh[0] + bbox_xywh[2] / 2, bbox_xywh[1] + bbox_xywh[3] / 2]
+    return center
```
Comment (felixNan) [inline lib/nanit_utils/utils_functions.py:15]: exists in python_tools

File: lib/nanit_utils/utils_functions.py [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    p_trans = np.array([p[0] + x_delta, p[1] + y_delta])
+    return p_trans
+
+
+def box_xyxy_to_xywh(box_xyxy):
+    min_x, max_x = min(box_xyxy[0], box_xyxy[2]), max(box_xyxy[0], box_xyxy[2])
+    min_y, max_y = min(box_xyxy[1], box_xyxy[3]), max(box_xyxy[1], box_xyxy[3])
+    box_xywh = [min_x, min_y, max_x - min_x,max_y - min_y]
+    return box_xywh
+
```
Comment (felixNan) [inline lib/nanit_utils/utils_functions.py:40]: exists in python_tools

File: lib/nanit_utils/utils_functions.py [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+
+def box_xywh_to_xyxy(box_xywh):
+    box_xyxy = [box_xywh[0], box_xywh[1], box_xywh[0] + box_xywh[2], box_xywh[1] + box_xywh[3]]
+    return box_xyxy
+
```
Comment (felixNan) [inline lib/nanit_utils/utils_functions.py:45]: exists in python tools

File: lib/nanit_utils/gen_coco_data.py (hunk 2 above)
Comment (tomer-algo) [reply lib/nanit_utils/gen_coco_data.py:1]: Moved to 'extract-and-gen-data' branch

File: lib/nanit_utils/gen_mpii_data.py (hunk 3 above)
Comment (tomer-algo) [reply lib/nanit_utils/gen_mpii_data.py:1]: Moved to 'extract-and-gen-data' branch

File: lib/nanit_utils/extract_data_from_batches.py (hunk 4 above)
Comment (tomer-algo) [reply lib/nanit_utils/extract_data_from_batches.py:1]: Moved to 'extract-and-gen-data' branch

File: lib/dataset/nanit_mpii.py (hunk 5 above)
Comment (tomer-algo) [reply lib/dataset/nanit_mpii.py:28]: it is also in use in the parent class
prefer not to change it

File: demo/demo_inference.py (hunk 1 above)
Comment (tomer-algo) [reply demo/demo_inference.py:247]: refactor script
including 'do_predict()'

File: lib/dataset/nanit_mpii.py (hunk 6 above)
Comment (tomer-algo) [reply lib/dataset/nanit_mpii.py:32]: using joints_mapping

File: lib/dataset/nanit_mpii.py (hunk 7 above)
Comment (tomer-algo) [reply lib/dataset/nanit_mpii.py:94]: done

File: lib/nanit_utils/utils_functions.py (hunk 9 above)
Comment (tomer-algo) [reply lib/nanit_utils/utils_functions.py:11]: removed

File: lib/nanit_utils/utils_functions.py (hunk 11 above)
Comment (tomer-algo) [reply lib/nanit_utils/utils_functions.py:40]: removed

File: lib/nanit_utils/utils_functions.py (hunk 12 above)
Comment (tomer-algo) [reply lib/nanit_utils/utils_functions.py:45]: removed

File: lib/nanit_utils/utils_functions.py (hunk 10 above)
Comment (tomer-algo) [reply lib/nanit_utils/utils_functions.py:15]: removed

**Review by felixNan — APPROVED** [review]

**Pull request merged**